src/sentry/api/endpoints/organization_member/__init__.py

```diff
@@ -27,32 +27,27 @@ def get_allowed_roles(
     request: Request,
     organization: Organization,
     member: OrganizationMember | None = None,
-) -> tuple[bool, Iterable[Role]]:
-    can_admin = request.access.has_scope("member:admin")
+) -> Collection[Role]:
+    if is_active_superuser(request):
+        return roles.get_all()
+    if not request.access.has_scope("member:admin"):
+        return ()
 
-    allowed_roles = []
-    if can_admin and not is_active_superuser(request):
-        if member:
-            acting_member = member
-        else:
-            try:
-                acting_member = OrganizationMember.objects.get(
-                    user=request.user, organization=organization
-                )
-            except OrganizationMember.DoesNotExist:
-                # This can happen if the request was authorized by an app integration
-                # token whose proxy user does not have an OrganizationMember object.
-                return can_admin, allowed_roles
+    if member:
+        acting_member = member
+    else:
+        try:
+            acting_member = OrganizationMember.objects.get(
+                user=request.user, organization=organization
+            )
+        except OrganizationMember.DoesNotExist:
+            # This can happen if the request was authorized by an app integration
+            # token whose proxy user does not have an OrganizationMember object.
+            return ()
 
-        if member and roles.get(acting_member.role).priority < roles.get(member.role).priority:
-            can_admin = False
-        else:
-            allowed_roles = acting_member.get_allowed_roles_to_invite()
-            can_admin = bool(allowed_roles)
-    elif is_active_superuser(request):
-        allowed_roles = roles.get_all()
-
-    return can_admin, allowed_roles
+    if member and roles.get(acting_member.role).priority < roles.get(member.role).priority:
```

Comment:
Yes -- in fact, it's relevant *only* if we grab the acting member from `OrganizationMember`. If we have a passed `member`, we enter this check but the priority comparison is always going to be false (the roles are equal because the members are the same).

The purpose of this check was opaque enough to trip me up in #33853 as well, so I've added a comment: https://github.com/getsentry/sentry/pull/33851/commits/49da6caaea99f3f2ea0763d2d042b13980a89bb5 Do you find that helpful?

Reply:
Ah nice, actually that makes sense.
Yup comment helps!
Thanks!